firedrake/functionspace.py

```diff
@@ -161,7 +161,8 @@ def DualSpace(mesh, family, degree=None, name=None, vfamily=None,
 
     # Support FunctionSpace(mesh, MixedElement)
     if type(element) is ufl.MixedElement:
-        return MixedFunctionSpace(element, mesh=mesh, name=name)
+        new = MixedFunctionSpace(element, mesh=mesh, name=name)
+        return impl.FiredrakeDualSpace.create(new, mesh)
```

Comment:
I'm doing it because otherwise calling `DualSpace` with a `MixedElement` returns a `FunctionSpace`  (actually a `WithGeometry`) and not a `FiredrakeDualSpace`. I'm 99% sure this is what causes the bug in #3234

Reply:
I also think that's the problem behind #3234. However, I don't think that's the right fix for it. I think the responsibility of `DualSpace` should be to call `MixedFunctionSpace` exactly as it is done for `FunctionSpace`. Getting the right mixed space is the responsibility of `MixedFunctionSpace`, so it should be fixed at that level. I think the line you want to fix is: https://github.com/firedrakeproject/firedrake/blob/4449d80b74bcafa46eae9743376ce15aaf8a93ad/firedrake/functionspace.py#L283

Reply:
I think it gets the right mixed space already (in that its the correct combination of subelements), it's just that it returns a primal space, not a dual space. How would `MixedFunctionSpace` know to return a dual space, is there a way to tell it? Or did you mean something else?

Reply:
> I think the line you want to fix is:
> 
> https://github.com/firedrakeproject/firedrake/blob/4449d80b74bcafa46eae9743376ce15aaf8a93ad/firedrake/functionspace.py#L283

I don't think this part of the code could be informed that we want a DualSpace, `spaces` here is just a list of elements. 

I agree with the current proposed solution of this PR